enterprise/server/remote_execution/containers/firecracker/firecracker.go

```diff
@@ -766,10 +800,24 @@ func (c *FirecrackerContainer) hotSwapWorkspace(ctx context.Context, execClient
 	if _, err := execClient.UnmountWorkspace(ctx, &vmxpb.UnmountWorkspaceRequest{}); err != nil {
 		return status.WrapError(err, "failed to unmount workspace")
 	}
-	chrootRelativeImagePath := filepath.Base(c.workspaceFSPath())
-	if err := c.machine.UpdateGuestDrive(ctx, workspaceDriveID, chrootRelativeImagePath); err != nil {
-		return status.InternalErrorf("error updating workspace drive attached to snapshot: %s", err)
+
+	if *enableNBD {
+		// Swap out the backing file for the workspace NBD.
+		if err := c.workspaceDevice.Close(); err != nil {
+			log.Warningf("Failed to close workspace nbd: %s", err)
+		}
+		store, err := blockio.NewMmap(c.workspaceFSPath())
+		if err != nil {
+			return status.WrapError(err, "initialize workspace NBD store")
+		}
+		c.workspaceDevice.Store = store
```

Comment:
this feels sketch -- you're reaching into a *nbdserver.Device and replacing a member of it. Probably would be better / safer to have a Replace()/Load() method on the Device object and use that, or to replace the Device object entirely here